### DISCUSSION TITLE
Tasmota IR doesnt send IR

### DISCUSSION THREAD

**Discussion by delaineS** [arendst/Tasmota#21277] [Support request]
Hi, I'm going crazy to get the ir to work for starting an air conditioner.
What I use: EPS8266, IR LED as per photo.
What I do: I point the air conditioner remote control at the receiver and I get this on console:
`11:38:51.130 RSL: RESULT = {"IrReceived":{"Protocol":"COOLIX","Bits":24,"Data":"0xB21F48","DataLSB":"0x4DF812","Repeat":0,"IRHVAC":{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off","SensorTemp":null}}}`

next, i send this:
`11:40:25.325 CMD: IRHVAC :{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off"}`

but the air conditioner does not respond
What i tested:
I tried to change Ir led and receiver, I tried to change more Esp8266, even testing an esp32 but it doesn't change. The LEDs work because I see the flash via the IR camera.
I also tried receiving the raw data and converting it but it doesn't work the same

I tried with a Philips TV but nothing happens...

what am I doing wrong?
![schematics](https://github.com/arendst/Tasmota/assets/92988734/2167e8ba-a138-4d24-ab36-6edbd9dbe773)

![322702591-36cc87c4-f5ef-4eef-a348-f00694d0fbaa](https://github.com/arendst/Tasmota/assets/92988734/c237aeab-b891-4941-8d31-c3b230334b86)

**Comment by reimerp** [comment]
Might be a Modulation Issue. Your IR Receiver does demodulation in HW and the Tasmota Lib modulates in Software AFAIR. Air Conditioner Protocols are quite complex, see https://github.com/crankyoldgit/IRremoteESP8266/wiki/Frequently-Asked-Questions#the-libraryexample-code-doesnt-understand-my-air-conditioner-remote-
(This is the Lib included in Tasmota) maybe you'll get some hints there.
I would go for a logic analyzer, sample the raw modulated signal (whith a simple IR Transistor not a receiver) from the original remote and the tasmota and compare. There are Libs available in eg pulseview to do decoding. What is remarkable that even Philips RC5/6 does not work with your setup, is this decoded?

**Comment by pkkrusty** [comment]
How far away is the IR LED from the AC? I've been playing with IR driver, and it has been working with every AC model I've tried. I even have a Coolix one with no issues.

Also, what is the exact command you put in the console? It should be `IRHVAC {"Vendor":"COOLIX","Model":-1,"Command":"Co...` with no colon

**Comment by mdk2412** [comment]
As far as I can remember my coolix unit also did not recognize the "full" IRHVAC JSON message. Try using only the minimum parameters, i.e. {"Vendor":"COOLIX","Mode":"Auto","Power":"On","Temp":22,"FanSpeed":"Auto"}.

**Comment by qingvsyu** [comment]
TRANSLATORS:
How to use mqtt in Homeassistant to control tasmota infrared transmission
![IMG_20240628_174323.jpg](https://github.com/arendst/Tasmota/assets/52342270/e3a06fd9-bf99-4c57-bf38-48dd24b9109d)

**Reply by Jason2866** [reply]
@qingvsyu Dont hijack this discussion with a non related question. AND If you have questions how HA stuff works ask in HA forums
